src/client/dfuse/dfuse_main.c

```diff
@@ -349,6 +351,28 @@ show_help(char *name)
 	    name, DAOS_VERSION);
 }
 
+static int
+check_fd_mountpoint(const char *mountpoint)
```

Comment:
Done.

Reply:
I'm sorry but I disagree, please don't ever set errno or use it like this.  It's a weird thread-safe hack that's used for lack of proper return codes in libc.

To do what Cedric is asking for the idiomatic way of doing this would be to have the function return a int with the 'return code' which would be defined as an int but return a negative 'daos errno', and then also have the function take a 'int *' for passing out the fd.

I'm not sure if it's worth the extra complexity if it's just going to be for logging in the parent however.

Reply:
I considered returning -daos_errno2der(errno) or -DER_INVAL on error, but it seemed kind of messy and maybe a misuse of the DAOS error codes.

Anyway, I'm happy to just omit this. I don't think the details of the error are all that useful to the user here.

Reply:
Fair enough for me.